packages/cli/generators/discover/index.js

```diff
@@ -289,6 +295,18 @@ module.exports = class DiscoveryGenerator extends ArtifactGenerator {
       );
       debug(`Discovered: ${modelInfo.name}`);
     }
+
+    if (this.options.optionalId) {
+      // Find id property
+      const idProperty = Object.entries(
+        this.artifactInfo.modelDefinitions[0].properties,
+      ).find(x => x[1].id === 1)[0];
```

Comment:
Can we future-proof this for composite IDs? Also, `id: true` is valid for single-IDs

E.g.

```suggestion
      ).find(x => x[1].id)[0];
```

Reply:
@achrinza  change has been made in the requested code.